nautobot/circuits/models.py

```diff
@@ -9,16 +10,54 @@
 from nautobot.core.models.generics import BaseModel, OrganizationalModel, PrimaryModel
 from nautobot.utilities.utils import serialize_object
 from .choices import CircuitTerminationSideChoices
-from .querysets import CircuitQuerySet
 
 
 __all__ = (
     "Circuit",
     "CircuitTermination",
     "CircuitType",
     "Provider",
+    "ProviderNetwork",
 )
 
+from ..utilities.querysets import RestrictedQuerySet
+
+
+@extras_features("custom_fields", "custom_links", "export_templates", "webhooks")
+class ProviderNetwork(PrimaryModel):
+    name = models.CharField(max_length=100)
```

Comment:
maybe support `slug` field? and use it instead of `pk`?

Reply:
Would you mind elaborate this? We could add a slug field, but until now, only organizational objects have a slug, e.g. Regions, Providers, Sites, Device Types, and not specific "implementations" of organizational objects.